src/editor/Editor.jsx

```diff
@@ -1,293 +1,344 @@
+    {label: 'Code Block', style: 'code-block'},
 ];
 
-var fontNameOptions = [
-	{ payload: 'Arial', text: 'Arial' },
-	{ payload: 'Georgia', text: 'Georgia' },
-	{ payload: 'Tahoma', text: 'Tahoma' },
-	{ payload: 'Courier New', text: 'Courier New' },
-	{ payload: 'Times New Roman', text: 'Times New Roman' },
-	{ payload: 'Verdana', text: 'Verdana' }
+// Inline Styles that are used in the editor and displayed in the toolbar
+const INLINE_STYLES = [
+    {label: 'Bold', style: 'BOLD'},
+    {label: 'Italic', style: 'ITALIC'},
+    {label: 'Underline', style: 'UNDERLINE'},
+    {label: 'Monospace', style: 'CODE'},
 ];
 
-var fontSizeOptions = [
-	{payload: 1, text: 'Smallest'},
-	{payload: 2, text: 'X-Small'},
-	{payload: 3, text: 'Small'},
-	{payload: 4, text: 'Normal'},
-	{payload: 5, text: 'Large'},
-	{payload: 6, text: 'X-Large'},
-	{payload: 7, text: 'Huge'}
-];
+/**
+ * Handle when displaying the block type controls and
+ * Set the styles of the controls if it is toggled or not
+ */
+const BlockStyleControls = (props) => {
+    let displayStyleButton = [];
+
+    // Loop thru the block types and setup the style button
+    BLOCK_TYPES.map(function (type) {
+
+        let className = props.theme.chamelEditorStyleButton;
+        if (type.style === props.blockType) {
+            className += ' ' + props.theme.chamelEditorActiveButton;
+        }
+
+        displayStyleButton.push(
+            <StyleButton
+                key={type.label}
+                className={className}
+                label={type.label}
+                onToggle={props.onToggle}
+                style={type.style}
+            />
+        );
+    })
+
+    return (
+        <div className={props.theme.chamelEditorControls}>
+            {displayStyleButton}
+        </div>
+    );
+};
+
+/**
+ * Handle when displaying the inline style controls and
+ * Set the styles of the controls if it is toggled or not
+ */
+const InlineStyleControls = (props) => {
+    let currentStyle = props.editorState.getCurrentInlineStyle();
+    let displayStyles = [];
+
+    // Loop thru the inline styles and setup the style button
+    INLINE_STYLES.map(function (type) {
+
+        let className = props.theme.chamelEditorStyleButton;
+        if (currentStyle.has(type.style)) {
+            className += ' ' + props.theme.chamelEditorActiveButton;
+        }
+
+        displayStyles.push(
+            <StyleButton
+                key={type.label}
+                className={className}
+                label={type.label}
+                onToggle={props.onToggle}
+                style={type.style}
+            />
+        );
+    })
+
+    return (
+        <div className={props.theme.chamelEditorControls}>
+            {displayStyles}
+        </div>
+    );
+};
+
+/**
+ * Style button component that will display the style button in the editor toolbar
+ */
+class StyleButton extends React.Component {
+    constructor() {
+        super();
+        this.onToggle = (e) => {
```

Comment:
Don't assign functions like this. Instead just put it below the constructor. like:

`
constructor() {
  ...
}

onToggle = (e) => { ....
`